src/components/AddVideoModal/AddVideoModalContainer.js

```diff
@@ -18,8 +18,21 @@ function AddVideoModalContainer({
   modalOpen,
   closeModal,
 }) {
-  const { selectedItems, setSelectedItems, handleSelectAdditionalItem } =
-    useArrayStateManager({ maxItems })
+  const {
+    selectedItems,
+    setSelectedItems,
+    handleSelectAdditionalItem,
+    handleRemoveItem,
+  } = useArrayStateManager({ maxItems })
+
+  const toggleVideoSelection = (video) => {
```

Comment:
This looks like an effective solution for the AddVideoModal but the bug is likely present in all components that are using the `useArrayStateManager` hook. It looks like there is a flaw in the ordering of the if statements in `handleSelectAdditionalItem`. Have a look and see if you can fix it there at the root of the issue.